### ISSUE TITLE
Dangling pointer during reduceToRegion

### ISSUE THREAD

**Issue by caballa** [seahorn/seahorn#242]
The error is an assertion violation so it can be only reproduced in Debug mode. 

Command to reproduce error: 

```
seapp --simplifycfg-sink-common=false -o ldv-cat-0006.pp.ms.bc --horn-mixed-sem --ms-reduce-main ldv-cat-0006.pp.bc 
```

[ldv-cat-0006.pp.bc.zip](https://github.com/seahorn/seahorn/files/3817380/ldv-cat-0006.pp.bc.zip)

**Comment by caballa** [comment]
PR #245 solves the error.

**Comment by caballa** [comment]
Fixed with commit 2110982
